vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html

```diff
@@ -143,6 +151,23 @@
             function onMouseLeave(e) {
                 $(this).removeClass("hova");
             }
+
+            function getSelectedSpace(blockId){
+	            $.ajax({
+		            url: "[(@{'/staff/module/'+${module.id}+'/slide/'+${slide.id}+'/space/'})]"+ blockId,
+		            type: 'GET',
+		            success: function(result) {
+		            	$("#selectedSpace").text(result.name);
+		            },
+		            error: function(data) {
+				            var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>We are sorry but something went wrong. Please try to delete again later.</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+				            $('.error').append(alert);
+				            $(".error").delay(4000).slideUp(500, function(){
+				            $(".error").empty();
+		            	});
+		            }
+	            });
+            }
             
             function onDoubleClick(e){
```

Comment:
This is just a terrible way of doing this. There should be a different handler for each type of block. The handlers can be added depending on the classes they have instead of doing a string comparing mega if (as it is now). Line 1412 would then add a different function that only handles space link blocks.

Reply:
And image blocks also get their own function.